### PR TITLE
Also add a reusable workflow for release

### PR DESCRIPTION
**Pull request by alexeagle**
Taken from https://github.com/bazel-contrib/rules-template/blob/0040d8ad2b0f84fb040117490682946dc6b479e8/.github/workflows/release.yml

### CODE REVIEW

File: .github/workflows/release_ruleset.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Reusable workflow that can be referenced by repositories in their .github/workflows/release.yaml.
+# See example usage in https://github.com/bazel-contrib/rules-template/blob/main/.github/workflows/release.yaml
```
Comment (gregmagolan) [inline .github/workflows/release_ruleset.yaml:2]: requirements:
- .github/workflows/release_prep.sh
- .github/workflows/ci.bazelrc

Comment (gregmagolan) [reply .github/workflows/release_ruleset.yaml:2]: Also `.bazelrc` technically

Comment (alexeagle) [reply .github/workflows/release_ruleset.yaml:2]: in theory we could even check for those files and give a nice error but no time today :)

Comment (alexeagle) [reply .github/workflows/release_ruleset.yaml:2]: PTAL

**Review by gregmagolan — APPROVED** [review]

**Pull request merged**